src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -72,27 +76,31 @@ static void uct_ib_mlx5dv_check_direct_nic(struct ibv_context *ctx,
     /* Create a DMA specific device from topology perspective */
     snprintf(dev_name, sizeof(dev_name), "%s_direct",
              uct_ib_device_name(&md->super.dev));
-    md->direct_nic_sys_dev = ucs_topo_get_sysfs_dev(dev_name, sys_path, 0);
+    sys_dev_dnic = ucs_topo_get_sysfs_dev(dev_name, sys_path, 0);
 
-    status = ucs_topo_sys_device_set_sys_dev_aux(dev->sys_dev,
-                                                 md->direct_nic_sys_dev);
+    status = ucs_topo_sys_device_set_sys_dev_aux(dev->sys_dev, sys_dev_dnic);
     if (status != UCS_OK) {
-        ucs_debug("ucs_topo_sys_device_set_sys_dev_aux failed: %s",
-                  ucs_status_string(status));
         goto out;
     }
 
     ucs_debug("%s: Direct NIC is supported sys_path='%s%s' "
               "sys_dev=%u sys_dev_aux=%u",
               uct_ib_device_name(&md->super.dev),
               (sys_path[0] != 0) ? "/sys" : "", sys_path, dev->sys_dev,
-              md->direct_nic_sys_dev);
+              sys_dev_dnic);
+    md->direct_nic_sys_dev = sys_dev_dnic;
     return;
+
 out:
-#endif
-    md->direct_nic_sys_dev = UCS_SYS_DEVICE_ID_UNKNOWN;
-    ucs_debug("%s: Direct NIC support disabled",
+    ucs_debug("%s: Direct NIC has been requested, but is not supported",
               uct_ib_device_name(&md->super.dev));
+#else
+    if (md_config->ext.direct_nic) {
```

Comment:
print the next debug line anyway, since if direct NIC is disabled and not supported by driver, now we are  not printing anything